src/crypto/sha/sha512.cpp

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <libp2p/crypto/sha/sha512.hpp>
+
+#include <openssl/sha.h>
+
+namespace libp2p::crypto {
+  common::Hash512 sha512(std::string_view input) {
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+    const auto *bytes_ptr = reinterpret_cast<const uint8_t *>(input.data());
+    return sha512(gsl::make_span(bytes_ptr, input.length()));
+  }
+
+  common::Hash512 sha512(gsl::span<const uint8_t> input) {
+    common::Hash512 out;
+    SHA512_CTX ctx;
+    SHA512_Init(&ctx);
```

Comment:
You need to check return codes.

Reply:
This is copy-paste from existing sha256 implementation. By the way, it is a good point. I will check what can we do here. Probably, outcome::result will help here too.

Reply:
As I saw in openssl implementation using Init, update and final without checking return code may be ok, since it is not checked in most cases. 
Moreover, there are some versions of openssl where signatures of these functions don't  return error code. 
However, implementations can be changed in the future, so pls consider adding this checks.

Reply:
I left TODOs in code